src/main/java/org/broadinstitute/hellbender/utils/SimpleInterval.java

```diff
@@ -225,10 +225,14 @@ public boolean overlaps( final Locatable other ) {
       * This is the same as plain overlaps if margin=0.
       *
       * @param other interval to check
-      * @param margin how many bases may be between the two interval for us to still consider them overlapping.
+      * @param margin how many bases may be between the two interval for us to still consider them overlapping; assumed non-negative
       * @return true if this interval overlaps other, otherwise false
+      * @throws IllegalArgumentException if margin is negative
       */
      public boolean overlapsWithMargin(final Locatable other, final int margin) {
+         if ( margin < 0 )
```

Comment:
Add curly braces around if body -- omitting curly braces for single-line ifs can lead to bugs later down the line.

Reply:
Also, this check should come after the checks for null, otherwise `other.toString()` could cause a `NullPointerException`

Reply:
done with braces

Reply:
for moving checking after null.
I thought about it, but did the margin check before null because I think if a margin is provided with negative value, that indicates a logic error has been made upstream so should warn about that.
But I now check for null on `other` in the error message. 
Do you agree?